test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  void CallParseConfiguration(MetadataAgentConfiguration& config, std::istream& stream) {
+    config.ParseConfiguration(stream);
+  }
+
+  static void SetUpTestCase() {}
+  static void TearDownTestCase() {}
+
+  void TestDefaultConfig(MetadataAgentConfiguration& config) {
```

Comment:
This doesn't depend on any state in the fixture and only calls public methods. Is there any reason to not keep it as a standalone top-level function?

Reply:
No particular reason - moved out of the class.